sdk/translation/azure-ai-translation-document/azure/ai/translation/document/aio/_operations/_patch.py

```diff
@@ -48,11 +54,144 @@
+    ):
+        """
+        :meta private:
+        """
+        (
+            client,
+            initial_response,
+            deserialization_callback,
+        ) = polling_method.from_continuation_token(continuation_token, **kwargs)
+
+        return cls(client, initial_response, deserialization_callback, polling_method)
+
+
+class AsyncDocumentTranslationLROPollingMethod(AsyncLROBasePolling):
+    """A custom polling method implementation for Document Translation."""
+
+    def __init__(self, *args, **kwargs):
+        self._cont_token_response = kwargs.pop("cont_token_response")
+        super().__init__(*args, **kwargs)
+
+    @property
+    def _current_body(self) -> TranslationStatus:
+        try:
+            return TranslationStatus(self._pipeline_response.http_response.json())
+        except Exception:  # pylint: disable=broad-exception-caught
+            return TranslationStatus()  # type: ignore[call-overload]
+
+    def _get_id_from_headers(self) -> str:
+        return (
+            self._initial_response.http_response.headers["Operation-Location"]
+            .split("/batches/")[1]
+            .split("?api-version")[0]
+        )
+
+    def finished(self) -> bool:
+        """Is this polling finished?
+
+        :return: True/False for whether polling is complete.
+        :rtype: bool
+        """
+        return self._finished(self.status())
+
+    @staticmethod
+    def _finished(status) -> bool:
+        if hasattr(status, "value"):
+            status = status.value
+        return str(status).lower() in _FINISHED
+
+    @staticmethod
+    def _failed(status) -> bool:
+        if hasattr(status, "value"):
+            status = status.value
+        return str(status).lower() in _FAILED
+
+    def get_continuation_token(self) -> str:
+        if self._current_body:
+            return self._current_body.id
+        return self._get_id_from_headers()
+
+    # pylint: disable=arguments-differ
+    def from_continuation_token(self, continuation_token: str, **kwargs: Any) -> Tuple:  # type: ignore[override]
+        try:
+            client = kwargs["client"]
+        except KeyError as exc:
+            raise ValueError("Need kwarg 'client' to be recreated from continuation_token") from exc
+
+        try:
+            deserialization_callback = kwargs["deserialization_callback"]
+        except KeyError as exc:
+            raise ValueError("Need kwarg 'deserialization_callback' to be recreated from continuation_token") from exc
+
+        return client, self._cont_token_response, deserialization_callback
+
+    async def _poll(self) -> None:
+        """Poll status of operation so long as operation is incomplete and
+        we have an endpoint to query.
+
+        :raises: OperationFailed if operation status 'Failed' or 'Canceled'.
+        :raises: BadStatus if response status invalid.
+        :raises: BadResponse if response invalid.
+        """
+        while not self.finished():
```

Comment:
Just noticing this and it looks like a bug, please check the other patched polling class too.

```suggestion
        if not self.finished():
```

Reply:
Updated